elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metric/functions/DailyAverageOverMonth.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.queryengines.sql.metric.functions;
+
+import com.yahoo.elide.core.exceptions.InternalServerErrorException;
+import com.yahoo.elide.datastores.aggregation.query.ColumnProjection;
+import com.yahoo.elide.datastores.aggregation.query.TimeDimensionProjection;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.metric.SQLMetricFunction;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.query.SQLQueryTemplate;
+import com.yahoo.elide.datastores.aggregation.time.TimeGrain;
+import com.yahoo.elide.request.Argument;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Daily average value of a metric monthly.
+ * Would create a query to calculate the daily sum, and then create a query to calculate the monthly average.
+ */
+public class DailyAverageOverMonth extends SQLMetricFunction {
+    public DailyAverageOverMonth() {
+        super(
+                "dailyAvgOverMonth",
+                "daily average over month",
+                "daily average of a metric over months",
+                "",
+                Collections.emptySet());
+    }
+
+    @Override
+    public SQLQueryTemplate resolve(Map<String, Argument> arguments,
+                                    String alias,
+                                    Set<ColumnProjection> dimensions,
+                                    TimeDimensionProjection timeDimension) {
+        try {
+            SQLQueryTemplate subQuery = SqlSum.class.newInstance().resolve(
```

Comment:
The linter will complain this is unused.  You might want to comment it out for now and add a TODO.